database/transaction.rs

```diff
@@ -322,62 +321,19 @@ impl<D: DurabilityClient> TransactionSchema<D> {
 
         let type_manager = Arc::into_inner(self.type_manager).expect("Failed to unwrap Arc<TypeManager>");
         drop(type_manager);
+        (profile, Ok((self.database, snapshot)))
+    }
+    
+    pub fn commit(mut self) -> (TransactionProfile, Result<(), SchemaCommitError>) {
```

Comment:
Make this commit method delegate to database's `schema_commit`.
TODO: remove this method and update the test accordingly on a follow-up PR